src/umpire/resource/FileMemoryResource.cpp

```diff
@@ -0,0 +1,107 @@
+//////////////////////////////////////////////////////////////////////////////
+// Copyright (c) 2016-20, Lawrence Livermore National Security, LLC and Umpire
+// project contributors. See the COPYRIGHT file for details.
+//
+// SPDX-License-Identifier: (MIT)
+//////////////////////////////////////////////////////////////////////////////
+#ifndef UMPIRE_FileMemoryResource_INL
+#define UMPIRE_FileMemoryResource_INL
+
+#include "umpire/resource/FileMemoryResource.hpp"
+
+#include "umpire/util/Macros.hpp"
+
+#include <errno.h>
+#include <string.h>
+#include <fcntl.h>
+#include <ctime>
+#include <sys/mman.h>
+#include <unistd.h>
+#include <utility>
+#include <stdlib.h>
+#include <sys/mman.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+
+namespace umpire {
+namespace resource {
+
+int FILE_COUNTER = 0;
+
+FileMemoryResource::FileMemoryResource(
+    Platform platform, 
+    const std::string& name,
+    int id,
+    MemoryResourceTraits traits) :
+  MemoryResource(name, id, traits),
+  m_platform{platform},
+  m_size_map{}
+{
+} 
+
+void* FileMemoryResource::allocate(std::size_t bytes)
+{
+  if (bytes <= 0) { UMPIRE_ERROR( "Bytes Requested Error: Bytes size is 0"); }
+
+  // Setting File Name And Opening the files
+  std::stringstream SS;
+  SS << "./umpire_mem_" << getpid() << FILE_COUNTER;
+  FILE_COUNTER++;
+
+  int fd = open(SS.str().c_str(), O_RDWR | O_CREAT | O_LARGEFILE, S_IRWXU);
+  if (fd == -1) { UMPIRE_ERROR("Opening File Failed: " << strerror(errno)); }
+
+  // Setting Size Of Map File
+  std::size_t SIZE = bytes / sysconf(_SC_PAGE_SIZE);
```

Comment:
For clarity, may I suggest:
```suggestion
  std::size_t num_pages = bytes / sysconf(_SC_PAGE_SIZE);
  std::size_t rounded_bytes;   // Round up bytes to multiple of page size
```